### ISSUE TITLE
Trino with elastic : how to search nested objects?

### ISSUE THREAD

**Issue by gilbob20000** [trinodb/trino#8050]
I'm new to trino and I'm trying to use it to query nested objects in elastic search.

This is my mapping in elasticsearch:

{
"product_index": {
   "mappings": {
      "properties" :{
          "id" : { "type" : "keyword"},
          "name" { "type" : "keyword"},
          "linked_products" :{
               "type": "nested",
               "properties" :{
                   "id" : { "type" : "keyword"}
                }
           }
        }
      }
    }
}
I need to perform a query on the id field under linked_products . what is the syntax in trino to perform a query on the id field? Do I need to use special definitions on the target index mapping in elastic to map the nested section for trino?

**Comment by JonathanBennett** [comment]
I tested using the solution from this issue (https://github.com/trinodb/trino/issues/2441) which is documented here https://trino.io/docs/current/connector/elasticsearch.html#array-types and this worked for me!

**Comment by gilbob20000** [comment]
Hi Jonathan
Can you please send the additional elastic mapping and the query syntax that you used?
Thank you very much!

**Comment by gilbob20000** [comment]
I will try to add some clarifications to my question.
 
We are trying to query the data according to the id field.
This is the query in Elastic:

    get product_index/_search
    { 
    "query": {
    "nested" : {
    "path" : "linked_products",
    "query": {
    "bool": {
    "should" : [
    { "match" : {"linked_products.id" :123}}
    ]
    }
    }
    }
    }
    }

We tried to query the id field  in 2 ways:

1) Trino query - 
When we try to query according to the id field the Pushdown to elastic doesn't
happen and the connector retrieve all the documents to trino (this only happens with queries on nested documents).

2) send es-query from trino to elastic:
SELECT * FROM es.default."<index>$query:<es-query>"
It works but when we are trying to retrieve id's with many documents we got timeout from the elastic client.
I don't understand from the documentation if it is possible to perform scrolling when are using es-query to avoid the timeout problem.
